src/main/java/filter/Parser.java

```diff
@@ -176,44 +172,42 @@ private FilterExpression parseQualifier(Token token) {
         // Strip the : at the end, then trim
         qualifierName = qualifierName.substring(0, qualifierName.length() - 1).trim();
 
-        return parseQualifierContent(qualifierName, false);
+        QualifierType type = QualifierType.parse(qualifierName).orElse(QualifierType.FALSE);
+        return parseQualifierContent(type, false);
     }
 
-    private FilterExpression parseQualifierContent(String qualifierName, boolean allowMultipleKeywords) {
-        if (isSortQualifier(qualifierName)) {
+    private FilterExpression parseQualifierContent(QualifierType qualifier, boolean allowMultipleKeywords) {
```

Comment:
Just for clarity: can we use `type` or `qualifierType` here instead of `qualifier` for variable name? Especially that we have a `Qualifier` class, this might confuse people as to its type.